odoo/models.py

```diff
@@ -4283,11 +4285,12 @@ def get_trans(field, old, new):
                 vals_list = []
                 for vals in Translation.search_read(domain):
                     del vals['id']
-                    del vals['source']      # remove source to avoid triggering _set_src
                     del vals['module']      # duplicated vals is not linked to any module
                     vals['res_id'] = target_id
+                    if not callable(field.translate):
+                        vals['src'] = new_wo_lang[name]
```

Comment:
Why?

Reply:
Because it was empty before, we used to set `src` only for the `vals['lang'] == old.env.lang` scenario but it must be set for all languages.